src/genie/libs/parser/asa/tests/ShowRoute/cli/equal/golden_output_4_expected.py

```diff
@@ -0,0 +1,84 @@
+expected_output = {
+    "vrf": {
+        "default": {
+            "address_family": {
+                "ipv4": {
+                    "routes": {
+                        "0.0.0.0/0": {
+                            "candidate_default": True,
+                            "active": True,
+                            "route": "0.0.0.0/0",
+                            "source_protocol_codes": "S",
+                            "source_protocol": "static",
+                            "metric": 0,
+                            "route_preference": 1,
+                            "next_hop": {
+                                "next_hop_list": {
+                                    1: {
+                                        "index": 1,
+                                        "next_hop": "104.225.199.193",
+                                        "outgoing_interface_name": "Outside-Port-Channel"
+                                    }
+                                }
+                            }
+                        },
+                        "1.1.1.0/24": {
+                            "candidate_default": False,
+                            "active": True,
+                            "date": "2d19h",
+                            "route": "1.1.1.0/24",
+                            "source_protocol_codes": "B",
+                            "source_protocol": "bgp",
+                            "metric": 0,
+                            "route_preference": 20,
+                            "next_hop": {
+                                "next_hop_list": {
+                                    1: {
+                                        "index": 1,
+                                        "next_hop": "10.98.126.1"
+                                    }
+                                }
+                            }
+                        },
+                        "4.33.104.119/32": {
+                            "candidate_default": False,
+                            "active": True,
+                            "date": "2d19h",
+                            "route": "4.33.104.119/32",
+                            "source_protocol_codes": "B",
+                            "source_protocol": "bgp",
+                            "metric": 0,
+                            "route_preference": 20,
+                            "next_hop": {
+                                "next_hop_list": {
+                                    1: {
+                                        "index": 1,
+                                        "next_hop": "10.98.126.1"
+                                    }
+                                }
+                            }
+                        },
+                        "0.0.0.0/0-1": {
```

Comment:
This seems weird. I would suggest to use index here.

Reply:
That's [exactly what I did for my first PR](https://github.com/CiscoTestAutomation/genieparser/pull/572) it is a much better solution. However: 

> all changes need to be backwards compatible

Reply:
Instead of closing the original PR, you should have addressed the comments on it. This way, it is easier to keep track of comments and requests. I am asking @LukasMcClelland to help with the review since he also reviewed your first PR.

Reply:
I was interested in quickly pivoting to a different model because I have a limited amount of time to work through projects. The original PR is mentioned on the issue, which is linked above and contains many valuable insights into the ongoing work.

Reply:
@domachad Some revelations were made in the issue which exposed the appropriate path forward for handling this edge case. Unless you have strong opinions otherwise I'm going to close this PR and open another so as to keep the commit history clean.

Reply:
@thismatters Ok, open a new PR, but please add @LukasMcClelland and me as reviewers